lightning/src/ln/channelmanager.rs

```diff
@@ -4075,6 +4098,13 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 								});
 							}
 
+							if let Ok(mut pending_events_lock) = self.pending_events.lock() {
+								pending_events_lock.push(events::Event::ChannelClosed {
+									channel_id: *channel_id,
+									reason: ClosureReason::HolderForceClosed
```

Comment:
Confused, isn't this a cooperative close?

Reply:
Yeah I was confused by the function name `maybe_generate_initial_closing_signed`